### PR TITLE
[sdc] update sdc after PD review

### PR DESCRIPTION
**Pull request by meisnere**
Please review the two files, confirm them and use it for future sdc edits.

chip_earlgrey_asic.ot.sdc replaces the previous chip_earlgrey_asic.sdc

### CODE REVIEW

**Comment by andreaskurth** [comment]
Thanks @meisnere! Is there a reason why this SDC file isn't named `chip_earlgrey_asic.sdc`? That would provide a diff compared to what is currently in the repo

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:1235]: Has this happened?

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:1203]: This should be removed.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:1167]: This constraint is commented out, should it be removed instead?

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:1105]: Has this approval happened?

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:984]: This constraint is commented out and could be removed, no?

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:801]: Has this been approved?

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:1020]: This should be removed.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:615]: This should be removed.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:490]: This should be removed.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:371]: ?

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:206]: Remove?

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc:192]: remove?

**Review by vogelpi — COMMENTED** [review]
Thanks @meisnere ! This is a lot of constraints to review, I did a first pass and left some feedback.

File: hw/top_earlgrey/syn/ot.sdc_setup.tcl
Comment (vogelpi) [reply hw/top_earlgrey/syn/ot.sdc_setup.tcl:7]: Thanks for adding the header @meisnere .

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc]: Thanks for implementing that change @meisnere .

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc]: This comment seems to be gone in the latest version and I assume the issue has been resolved.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc]: The comment has been updated to say that this constraint got added based on interaction between Leonid and @a-will . This looks okay to me.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc]: This commented out constraint is still part of the last revision. It's fine to leave them in to not further increase the size of the diff (these comments were already part of the SDC used for ES).

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc]: The comment has been updated to say that this constraint got added based on interaction between Leonid and @a-will . This looks okay to me.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc]: This comment is still in the updated version. And I'll leave an additional comment there.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:788]: @meisnere @OTshimeon , I've asked for clarification on the status of this in the previous version of the PR. Who needs to approve this? Based on what assumptions / findings / discussions has this been added? Please comment.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc]: This TODO has already been in the previous SDC file used for ES. @meisnere , @OTshimeon please comment on whether this is actually needed for the production tapeout.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (vogelpi) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:1552]: This is the pad used for SPI TPM CSB and this may be the reason for using different values here compared to the other pads below. We need to review these constraints in a separate document.

**Review by vogelpi — COMMENTED** [review]
Thanks @meisnere and @OTshimeon for updating the PR. I did another pass and  will now set up a shared document to continue the review of some items internally. To proceed with this PR, I need input from you on the following things:
- Please comment on the remaining open questions.
- Please run `util/fix_trailing_whitespace.py hw/top_earlgrey/syn/chip_earlgrey_asic.sdc` to remove trailing whitespaces. The PR currently fails lint and we can't merge it like this. For details, see the "checks" section below.
- The constraints we've checked and approved last TUE in a meeting (they were in a different file previously) seem not to be part of this PR. Please explain why and what the planned course of action is here. My understanding was that these constraints need to be moved to this SDC file as well. FYI @moidx .

File: hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc]: The comment has been updated to say that this constraint came from @a-will . This looks okay to me.

**Comment by meisnere** [comment]
Add the last few missing constraints - line 1664 and below.

**Comment by a-will** [comment]
> I also don't see any constraints for trst_n which is connected to the IO4 pad. Should we have some? I don't see any obvious constraints around other resets so I presume this is fine but wanted to check.

Nope. I believe there is no need for a timing requirement for nTRST. It's an async reset, and the user is required to assert it only when TCK is inactive. Because the acceptable reset and guard times are on the order of hundreds of milliseconds, we give the tool freedom to place and route.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (hcallahan-lowrisc) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:684]: Nit. This comment appears stale, given the addition of IOA2 below?

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (hcallahan-lowrisc) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:1311]: Nit. duplicates. The sections L1311-1353 and L1355-1397 are identical?

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (hcallahan-lowrisc) [inline hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:933]: The `else` case (and the one above on L909) do not have output_delay constraints. All other `$spec_constr` if/else blocks duplicate the commands within each case. Is that intentional?

**Review by hcallahan-lowrisc — COMMENTED** [review]
I reviewed the spi constraints section (L289 to 1569) and have no major concerns.
Most of my comments were just nits/suggestions, which can mostly be ignored to keep the diff small if needed.

The only non-nit was about different output_delay constraints for some of the `if {$spec_constr}` blocks, which probably just needs a comment to articulate why.

**Review by alees24 — COMMENTED** [review]
I've been through the USB section of the SDC file (lines [53:121]) and checked all of the numbers and signals and I'm happy that the constraints are good. The critical signals have appropriate time constraints, with margins.
The `cio_usb_sense_i` signal changes much less frequently than the other inputs and is not at all timing-critical, but over-constraining it is harmless.
The explanatory commenting in the USB section should be helpful to others, thank you.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:1552]: @andreaskurth and I discussed this. The reason for the differences here is that this pad is an open drain pad which has just one bit for drive strength. I'll add a comment.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:1311]: Thanks for flagging this @hcallahan-lowrisc . I'll remove the duplicate in the next push.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.sdc]: I've now removed this comment as @hcallahan-lowrisc carefully reviewed all SPI related constraints (see https://github.com/lowRISC/opentitan/pull/24221#pullrequestreview-2288609680).

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:684]: I did have another look at this but could not find a reason why IOA2 should get these constraints. IOA2 doesn't seem to be related to SPI TPM at all. It's used a GPIO and in some cases we use it for SPI Host 1. But SPI Host 1 is not used for passthrough / TPM mode. This is only done for SPI device and SPI Host 0. So it's not clear to me why this is needed actually.

**Comment by vogelpi** [comment]
Thanks @hcallahan-lowrisc , @a-will , @GregAC , @alees24 for your reviews!

I've included the obvious fixes and @andreaskurth appended another commit with additional constraints seperately discussed and approved by the PD team. We will merge this once CI passes.

We will discuss the remaining questions and opens with the PD team in a follow-up.

**Review by vogelpi — APPROVED** [review]

**Review by andreaskurth — APPROVED** [review]
Agree with @vogelpi. Thx all!

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:684]: The PD team got back to me about this: Some time in the past, IOA2 was determined to be used as chip select for quad SPI. During the tapeout, it was agreed that IOA7 should be the primary chip select target whereas IOA2 should remain as a secondary opportunistic target. I'll add a comment into the SDC to document this.

File: hw/top_earlgrey/syn/chip_earlgrey_asic.sdc
Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:932]: Yes, it actually is intentional. The PD provided feedback on this. The `spec_constr == 1` case in these two particular cases verifies the fast passthrough mode where the direct timing paths from an upstream host to a downstream Flash device is checked. The SPI Device is then defined as input and the SPI Host as output.

In contrast, the `spec_constr == 0` case verifies the slow passthrough mode where the direct timing paths from a downstream Flash to an upstream host are checked. Then, the SPI device is defined as output and the SPI Host is defined as input. In this case, SPI Device gets no input delay constraints and SPI Host not output delay constraints.

I'll document this in the SDC.

Comment (vogelpi) [reply hw/top_earlgrey/syn/chip_earlgrey_asic.sdc:932]: Ah, and the PD team also confirmed that both constraint sets got verified for the tapeout.

**Pull request merged**
